torch/_inductor/config.py

```diff
@@ -943,7 +943,8 @@ class rocm:
 
     # Enable for CDNA3 only for now
     # Processor name reference: https://llvm.org/docs/AMDGPUUsage.html#processors
-    supported_arch: Set[str] = {"gfx940", "gfx941", "gfx942"}
+    # Keep it ordered, unordered set can cause spurious inductor cache misses
+    supported_arch: Dict[str, None] = dict.fromkeys(["gfx940", "gfx941", "gfx942"])
```

Comment:
We should just use https://github.com/pytorch/pytorch/pull/130003 once it's landed..

Reply:
oooh originally I found `OrderedSetHolder` in `codecache.py` but I was worried about circular imports. Once the torch.utils one lands that will be great.

Reply:
updated